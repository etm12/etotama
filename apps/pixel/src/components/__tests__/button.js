import * as React from 'karet';
import { mount } from 'enzyme';
import Button from '../button';

describe('Button', () => {
  test('mounts without crashing', () => {
    const result = mount(<Button />);
    expect(result).toMatchSnapshot();
  })

  test('supports custom action', () => {
    const fn = jest.fn();
    const result = mount(<Button action={fn} />);
    result.find('button').simulate('click');

    expect(fn).toBeCalled();
  });

  test('set height properly as rems', () => {
    const result = mount(<Button height={6} />);

    expect(result).toMatchSnapshot();
    expect(result.find('button').prop('style')).toEqual({
      height: '6rem',
    });
  });
});
