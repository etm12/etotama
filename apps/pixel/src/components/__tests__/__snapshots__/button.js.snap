// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> mounts without crashing 1`] = `
<Button>
  <button
    className="c-button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": undefined,
        "height": undefined,
      }
    }
  />
</Button>
`;

exports[`<Button /> set height properly as rems 1`] = `
<Button
  height={6}
>
  <button
    className="c-button"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": undefined,
        "height": "6rem",
      }
    }
  />
</Button>
`;
